### PR TITLE
Flake8 rule C901 is about McCabe code complexity

### PR DESCRIPTION
**Pull request by cclauss**
Flake8 C901 is **_not_** about [`flake8-comprehensions`](https://pypi.org/project/flake8-comprehensions/) which are C4xx rules.

This misconfiguration has made us blind to increasing code complexity...  https://pypi.org/project/mccabe
```diff
- max-complexity = 15
+ max-complexity = 29
```
Related to:
* #2410

## Reviewer Checklist
- [x] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # Exclude the grpc generated code
 exclude = ./manim/grpc/gen/*, __pycache__,.git,
 per-file-ignores = __init__.py:F401
-max-complexity = 15
+max-complexity = 29
```
Comment (JasonGrace2282) [inline .flake8:5]: Why 29?

**Review by JasonGrace2282 — COMMENTED** [review]
Nice catch, just one question.

File: .flake8 (hunk 1 above)
Comment (cclauss) [reply .flake8:5]: https://results.pre-commit.ci/run/github/265122478/1712226727._i0WCFzjSKy7ADvbItyMbQ

**Review by behackl — APPROVED** [review]
Thanks!

**Pull request merged**